### DISCUSSION TITLE
Post `guid` vs the meta-key 'external-id' for storing the origin of external posts.

### DISCUSSION THREAD

**Discussion by Menrath** [akirk/friends#412] [Q&A] [answered]
@akirk I would like to understand why both are used for storing the origin of an external resource. 

https://github.com/akirk/friends/blob/199af50ae54ee82e8cc77b2311a7149fb1eea8b2/feed-parsers/class-feed-parser-activitypub.php#L1890-L1893

I would appreciate sharing your thoughts on this design decision back then!

**Comment by akirk** [comment, marked as answer]
The guid is the fallback when there is no external id since [it only is set for posts coming in via ActivityPub](https://github.com/akirk/friends/blob/199af50ae54ee82e8cc77b2311a7149fb1eea8b2/feed-parsers/class-feed-parser-activitypub.php#L943). The external id will always be the id, both for posts received via Create or Announce ([which funnels into handle_incoming_create](https://github.com/akirk/friends/blob/199af50ae54ee82e8cc77b2311a7149fb1eea8b2/feed-parsers/class-feed-parser-activitypub.php#L1096)), but for announces [the guid will be the actual URL](https://github.com/akirk/friends/blob/199af50ae54ee82e8cc77b2311a7149fb1eea8b2/feed-parsers/class-feed-parser-activitypub.php#L933-L936).

Since the Friends plugin can also receive posts through other sources like RSS, it could be that the external id is not set. And in that case, the URL is the fallback.
